File: app/src/main/java/app/rpgbuilder/activities/MainActivity.java
package app.rpgbuilder.activities;

import static android.Manifest.permission.POST_NOTIFICATIONS;

import static androidx.activity.result.contract.ActivityResultContracts.RequestMultiplePermissions;
import static androidx.activity.result.contract.ActivityResultContracts.StartActivityForResult;

import app.rpgbuilder.AutoSaveService;
import app.rpgbuilder.R;

import android.content.pm.PackageManager;

import android.graphics.Color;

import android.os.Build;
import android.os.Bundle;

import android.content.Intent;

import androidx.activity.result.ActivityResultLauncher;

import android.widget.TextView;

import androidx.annotation.RequiresApi;

import androidx.appcompat.app.AppCompatActivity;

import java.util.List;

public final class MainActivity extends NotificationBaseActivity {

    public MainActivity() {
        super(R.layout.main_layout);
    }

    @RequiresApi(api = Build.VERSION_CODES.TIRAMISU)
    @Override
    protected void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        /*
        final var permissionsLauncher = registerForActivityResult(new RequestMultiplePermissions(), permissions -> {
            if (permissions.containsValue(false))
                finish();
        });
        */

        setSupportActionBar(findViewById(R.id.toolbar));

        if (checkSelfPermission(POST_NOTIFICATIONS) == PackageManager.PERMISSION_DENIED)
            requestPermissions(new String[] {POST_NOTIFICATIONS}, APPLICATION_REQUEST_CODE);
            //permissionsLauncher.launch(new String[]{POST_NOTIFICATIONS});
    }

    @Override
    protected void onStart() {
        super.onStart();

        final var activityLauncher = registerForActivityResult(new StartActivityForResult(), result -> {
            final Intent intent = result.getData();
        });

        startService(new Intent(this, AutoSaveService.class));

        findViewById(R.id.start_button).setOnClickListener(view -> {
            final TextView self = (TextView) view;

            activityLauncher.launch(new Intent(this, OptionsActivity.class));
            self.setTextColor(Color.RED);
        });
    }

    @Override
    protected void onResume() {
        super.onResume();

        final TextView startButton = findViewById(R.id.start_button);
        startButton.setTextColor(Color.WHITE);
    }

    @Override
    protected void onPause() {
        super.onPause();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }
}

File: app/src/main/java/app/rpgbuilder/tmpkt/activities/MainActivity.kt
package app.rpgbuilder.tmpkt.activities

import app.rpgbuilder.R

import android.content.pm.PackageManager.PERMISSION_DENIED

import android.Manifest.permission.POST_NOTIFICATIONS

import android.content.Intent

import android.os.Build
import android.os.Bundle

import android.view.Menu

import android.widget.TextView

import androidx.activity.result.contract.ActivityResultContracts.StartActivityForResult

import androidx.annotation.RequiresApi

import androidx.appcompat.app.AppCompatActivity

private const val REQUEST_CODE = 101

class MainActivity : AppCompatActivity(R.layout.main_layout) {

    @RequiresApi(Build.VERSION_CODES.TIRAMISU)
    override fun onCreate(savedInstanceState: Bundle?): Unit {
        super.onCreate(savedInstanceState)

        if (checkSelfPermission(POST_NOTIFICATIONS) == PERMISSION_DENIED)
            requestPermissions(arrayOf(POST_NOTIFICATIONS), REQUEST_CODE)
    }

    override fun onStart(): Unit {
        super.onStart()

        findViewById<TextView>(R.id.start).setOnClickListener {
            it as TextView

            registerForActivityResult(StartActivityForResult()) {}
                .launch(Intent(this, OptionsActivity::class.java))
        }
    }

    override fun onRequestPermissionsResult(
        requestCode: Int,
        permissions: Array<out String>,
        grantResults: IntArray
    ): Unit {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)

        if (requestCode == REQUEST_CODE) {
            if (grantResults.contains(PERMISSION_DENIED)) {
                finish()
            }
        }
    }
}